### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by Ibizugbe**
In this PR, I have completed the current milestone of the project and kept to the given requirements.
- There are no linter errors.
- Project is professionally documented.
- The right git-flow practice has been used.
- RSpec has been added as a gem to my project.
- A request spec file has been created for all my controllers.
- for each action, the status is correct, the correct template has been rendered and the body includes the correct placeholder test.

### CODE REVIEW

File: spec/requests/post_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+RSpec.describe PostsController, type: :request do
+  describe 'Check post main page' do
+    before(:example) do
+      get '/users/1/posts'
+      expect(response).to render_template(:index)
```
Comment (DammyShittu) [inline spec/requests/post_controller_spec.rb:9]: - [ ] The `expect` should be in its own **`it`** block. Kindly remove it from here.

File: spec/requests/post_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    it 'Check the if the rendered template is correct' do
+      expect(response.body).to include('<p>this is a list of all posts</p>')
+    end
+  end
+
+  describe 'Check posts for specific user' do
+    before(:example) do
+      get '//users/1/posts/:id'
+      expect(response).to render_template(:show)
```
Comment (DammyShittu) [inline spec/requests/post_controller_spec.rb:27]: - [ ] The `expect` should be in its own **`it`** block. Kindly remove it from here.

File: spec/requests/user_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    it 'Checks if it render the correct template' do
+      expect(response).to render_template(:show)
+    end
+
+    it 'Check if it has the correct placeholder text' do
+      expect(response.body).to include('<p>This is a specified user</p>')
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/requests/user_controller_spec.rb:38]: - [ ] Some of your tests are failing. Kindly fix them.

  asew

File: spec/requests/post_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    end
+
+    it 'Check the posts page for a given user' do
+      expect(response).to render_template(:show)
+      expect(response.body).to include('<p>this is a specified post</p>')
+      expect(response.status).to be(200)
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/requests/post_controller_spec.rb]: - [ ] Some of your tests are failing. Kindly fix them.

  tui

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Ibizugbe,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/post_controller_spec.rb (hunk 1 above)
Comment (Ibizugbe) [reply spec/requests/post_controller_spec.rb:9]: Thanks a lot @DammyShittu  for this observation, I'll do well to fix it immediately👍.

File: spec/requests/post_controller_spec.rb (hunk 2 above)
Comment (Ibizugbe) [reply spec/requests/post_controller_spec.rb:27]: I will fix that immediately, thanks.

File: spec/requests/user_controller_spec.rb (hunk 3 above)
Comment (Ibizugbe) [reply spec/requests/user_controller_spec.rb:38]: Thanks for your observation @DammyShittu , everything is currently passing on my local computer 🤔 kindly
Screenshot (152)
 check the picture attached. I ran the test severally and they are passing.

File: spec/requests/post_controller_spec.rb (hunk 4 above)
Comment (Ibizugbe) [reply spec/requests/post_controller_spec.rb]: Thanks, @DammyShittu for taking out time to review my project, Kindly check the comment above, My tests are passing on my computer.

**Review by prolajumokeoni — APPROVED** [review]
Hi @Ibizugbe 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀  
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ibizugbe** [comment]
Thanks, @prolajumokeoni  for reviewing my project👍.

**Pull request merged**